Source/Session.swift

```diff
@@ -750,11 +768,17 @@ open class Session {
 
     // MARK: Perform
 
+
+    /// Perform `Request`.
+    ///
+    /// - Note: Called during retry.
+    ///
+    /// - Parameter request: The `Request` to perform.
     func perform(_ request: Request) {
         switch request {
-        case let r as DataRequest: perform(r)
-        case let r as UploadRequest: perform(r)
-        case let r as DownloadRequest: perform(r)
+        case let r as DataRequest: self.perform(r)
+        case let r as UploadRequest: self.perform(r)
+        case let r as DownloadRequest: self.perform(r)
```

Comment:
Do we actually need `self.` now?

Reply:
Nope, removed in https://github.com/Alamofire/Alamofire/pull/2890/commits/39aaa58af41d5d423c60d21613c64a1f5b75d0bd.